backend/pkg/handlers/videoSummaryHandler.go

```diff
@@ -0,0 +1,38 @@
+package handlers
+
+import (
+	"Learning-Mode-AI/pkg/services"
+	"encoding/json"
+	"log"
+	"net/http"
+)
+
+type VideoSummaryRequest struct {
```

Comment:
I am thinking instead of passing the transcript in every request we make, we just pass in the video id, and get the transcript from redis cache. Might not be the best idea to be passing full on transcripts from the frontend to the backend in the payload.

Reply:
Might make more sense if we can do that at an even lower level, at the ai service. So pass the video id to the ai service when asking it to generate summary, and the ai service checks redis.